src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_CollectionExpression.cs

```diff
@@ -660,6 +659,15 @@ private BoundExpression CreateAndPopulateArray(BoundCollectionExpression node, A
                     var rewrittenSpreadExpression = VisitExpression(spreadExpression);
                     return _factory.Call(rewrittenSpreadExpression, listToArrayMethod.AsMember((NamedTypeSymbol)spreadExpression.Type!));
                 }
+                else if (IsConvertibleToObject(arrayType.ElementType)
+                    && _factory.WellKnownMethod(WellKnownMember.System_Linq_Enumerable__ToArray) is { } linqToArrayMethodGeneric)
+                {
+                    var linqToArrayMethod = linqToArrayMethodGeneric.Construct([arrayType.ElementTypeWithAnnotations]);
+                    if (ShouldUseAddRangeOrToListMethod(spreadExpression.Type!, linqToArrayMethod.Parameters[0].Type, spreadElement.EnumeratorInfoOpt?.GetEnumeratorInfo.Method))
+                    {
+                        return _factory.Call(receiver: null, linqToArrayMethod, VisitExpression(spreadExpression));
```

Comment:
>VisitExpression [](http://example.com/codeflow?start=80&length=15)

Do we have a test case that fails if we don't rewrite `spreadExpression`?

Reply:
Added test 'SingleSpread_ToArray_MustLowerSpreadOperand'.